### PR TITLE
Show email on settings when using external auth

### PR DESCRIPTION
**Pull request by ShaneKilkelly**
Also defend server-side against the email address being set, when using an external auth source.

### CODE REVIEW

File: app/coffee/Features/User/UserController.coffee [hunk 1]
```diff
@@ -44,6 +44,7 @@ module.exports = UserController =
 
 	updateUserSettings : (req, res)->
 		user_id = AuthenticationController.getLoggedInUserId(req)
+		usingExternalAuth = !!(settings.ldap? or settings.saml?)
```
Comment (ShaneKilkelly) [inline app/coffee/Features/User/UserController.coffee]: Possibly not, just forcefully coercing to boolean. Without the `!!`, `usingExternalAuth` ends up as whatever the value of `settings.saml` or `settings.ldap` is. It'll be truthy for sure, but doesn't feel as clean.

Comment (ShaneKilkelly) [inline app/coffee/Features/User/UserController.coffee]: Actually, no, disregard, the `?` takes care of that, will remove `!!`

Comment (henryoswald) [inline app/coffee/Features/User/UserController.coffee]: wouldn't this always result in a true or false?

usingExternalAuth = settings.ldap? or settings.saml?

Comment (henryoswald) [inline app/coffee/Features/User/UserController.coffee]: maybe we should add a settings.usingExternalAuth = settings.ldap? or settings.saml?

**Pull request merged**